src/Akka.Hosting/ActorRegistry.cs

```diff
@@ -31,13 +54,53 @@ public interface IRequiredActor<TActor>
     /// <typeparam name="TActor">The type key of the actor - corresponds to a matching entry inside the <see cref="IActorRegistry"/>.</typeparam>
     public sealed class RequiredActor<TActor> : IRequiredActor<TActor>
     {
+        private readonly IReadOnlyActorRegistry _registry;
+
         public RequiredActor(IReadOnlyActorRegistry registry)
         {
-            ActorRef = registry.Get<TActor>();
+            _registry = registry;
         }
 
+        private IActorRef? _internalRef = null;
+
         /// <inheritdoc cref="IRequiredActor{TActor}.ActorRef"/>
-        public IActorRef ActorRef { get; }
+        public IActorRef ActorRef
+        {
+            get
+            {
+                // attempt 1 - used cached value
+                if (_internalRef != null)
+                    return _internalRef;
+
+                // attempt 2 - synchronously check the registry (fast path)
+                if (_registry.TryGet<TActor>(out _internalRef))
+                {
+                    return _internalRef;
+                }
+
+
+                throw new MissingActorRegistryEntryException(
```

Comment:
Try our best to give the users a clear signal as to why this failed and what they should do instead. We can't really solve the problem for them and the normal, sync `ActorRef` property works for 99.9% of cases so we don't want to remove that just to support an edge case.